.github/workflows/deploy.yml

```diff
@@ -0,0 +1,121 @@
+name: deploy
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - lab
+      - master
+      - published
+
+# these permissions are needed to interact with GitHub's OIDC Token endpoint.
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
+    steps:
+      -
+        name: Prepare
+        run: |
+          JEKYLL_ENV=development
+          DOCS_AWS_REGION=us-east-1
+          if [ "${{ github.ref }}" = "refs/heads/master" ]; then
+            DOCS_URL="https://docs-stage2.docker.com"
+            DOCS_AWS_IAM_ROLE="arn:aws:iam::710015040892:role/stage-docs-docker.github.io-20220816140248629900000003"
```

Comment:
Bit of a nit, but if you're gonna be updating this branch, perhaps you can add TODO's for those parts that are temporary, and to be replaced.